runtime/runtime/src/actions.rs

```diff
@@ -738,17 +753,43 @@ fn validate_delegate_action_key(
     result: &mut ActionResult,
 ) -> Result<(), RuntimeError> {
     // 'delegate_action.sender_id' account existence must be checked by a caller
-    let mut access_key = match get_access_key(
-        state_update,
-        &delegate_action.sender_id,
-        &delegate_action.public_key,
-    )? {
+    let signer_id = &delegate_action.sender_id;
+    let public_key: &PublicKey = &delegate_action.public_key;
+
+    #[cfg(feature = "protocol_feature_eth_implicit")]
+    let mut access_key = get_access_key(state_update, &signer_id, &delegate_action.public_key)?;
+    #[cfg(not(feature = "protocol_feature_eth_implicit"))]
+    let access_key = get_access_key(state_update, &signer_id, &delegate_action.public_key)?;
+
+    #[cfg(feature = "protocol_feature_eth_implicit")]
+    if checked_feature!("stable", EthImplicit, apply_state.current_protocol_version) {
+        // Access key is missing, but in the case of a transaction from ETH-implicit account
+        // there is a possibility that full access key will be created now.
+        if access_key.is_none() && signer_id.get_account_type() == AccountType::EthImplicitAccount {
+            if derive_account_id_from_public_key(public_key) == *signer_id {
+                // TODO What about increasing storage usage for that account, because we added access key?
```

Comment:
We add access key here. That increases storage for that account. Would it be correctly handled later?

Reply:
We're having discussions with Bowen that instead of adding a FAK on the first transaction, we could associate a smart contract with the account from the time it is first created (no storage cost is needed because this will be a static Wasm blob at the protocol level). This contract could be used to handle adding keys instead of doing it at the protocol level.

Perhaps we should have a call about this some time after nearcon to discuss in more detail.

Reply:
@birchmd and @bowenwang1996 Where possible can you please invite Adam and myself to such discussions, or at the very least share updates in timely manner? Participating in the discussions would be much better as we would get more context and understanding of the problem and tradeoffs involved as well contribute to the discussion. 

Can you please advise how to proceed with this PR? Shall we put it on hold until we have all the details nailed down? It doesn't make much sense to go back and forth implementing all possible ideas and throw them away immediately and it must be rather frustrating.

Reply:
Yes, I understand the frustration of trying to implement something when the design is still changing. I think we should put this PR on hold until the design is finalized. I certainly want both @wacban and @staffik to participate in discussions where the design is agreed upon!